include/nbl/video/IGPUCommandBuffer.h

```diff
@@ -227,6 +227,25 @@ class NBL_API IGPUCommandBuffer :
             for (uint32_t i = _first + 1u; i < IGPUPipelineLayout::DESCRIPTOR_SET_COUNT; ++i)
                 _destPplnLayouts[i] = nullptr;
     }
+
+    uint32_t m_resetCount = 0;
+
+    virtual void resetCommon()
```

Comment:
make this function non virtual
have another virtual function called `void releaseResourcesToPool`